specifications/drtm-handler-and-table.rst

```diff
@@ -0,0 +1,310 @@
+supported by TrenchBoot. This will provide a well-defined interface for DLE
+Handler implementors to follow and bootloader to comply to ensure
+interoperability between implementations.
+
+Acronyms
+--------
+
+DCE - Dynamic Configuration Environment (Intel ACM/AMD SLB)
+DLE - Dynamic Launch Event (Intel GETSEC[SENTER]/AMD SKINIT)
+DLE Handler - Dynamic Launch Event Handler
+DLME - Dynamic Launch Measured Environment (Operating System/Hypervisor)
+DRTM - Dynamic Root of Trust Measurement
+SL Entry - Secure Launch Entry (Secure Launch Stub for Linux)
+
+Platform Requirements
+---------------------
+
+1 - x86 Platforms
+1.1 - The DLE Handler will be invoked with the CPU in 64bit long mode 
+1.2 - The DRTM Table will be passed to the DLE Handler in the RDI CPU register
+
+2 - Arm Platforms
+2.1 - TBD
+
+
+DRTM Table Specification
+========================
+
+This specification details the construction of the DRTM Table and how that
+table will be passed to a Secure Launch entry point for each supported hardware
+platform.
+
+Overview
+--------
+
+Like most hardware capabilities implemented by different manufactures, each
+platform has its own unique way of interacting with the capability. Dynamic
+Launch is no different in this manner. The TrenchBoot project is striving to
+provide a unified approach to using Dynamic Launch across different platforms.
+One aspect of Dynamic Launch that varies across platforms is the launch related
+data. For example, on Intel there is a series of Intel defined configuration
+data structures that must be present as well as a user-defined data structure.
+Whereas on AMD, only the layout of an AMD Secure Loader Block is defined and
+thus any configuration structures are SLB implementation specific.
+
+To provide a unified experience in passing configuration and other meta-data to
+a TrenchBoot Secure Launch entry point for a kernel, this specification
+provides the platform-agnostic DRTM Table along with how it will be passed for
+each platform supported.
+
+
+DRTM Table
+----------
+
+The DRTM Table provides a platform-agnostic, standard format for providing
+information to the DL Handler and for passing DRTM relevant information across
+the DLE to be available for use by the DCE and the DLME. The table will be
+initialized by a bootloader and any subsequent bootloaders in the boot chain
+may append entries to the table. While the table is designed to be agnostic,
+the reality of DRTM hardware will drive a difference in the construction and
+setup of the table.
+
+Platform Requirements
+^^^^^^^^^^^^^^^^^^^^^
+
+1 - General Requirements
+1.1 - The DRTM Table must be constructed within the 4G boundary
+1.2 - The DRTM Table should be in contiguous physical memory
+1.2.1 - A preallocated fixed size table is supported through the use of the
+        table_max_size field
+1.3 - The DRTM Table should start on a four-byte boundary
+1.4 - The DRTM Table tags 0x0000 - 0x0007 are reserved
```

Comment:
Why are those reserved?

Reply:
We realized we did not have a solid reason for reserving them so they no longer are. The one thing to note is that we reserve 0x0000 for the SLR_ENTRY_INVALID. We do this in case there are tables e.g. for padding that no one will ever read or if software doesn't properly initialize a table.